services/blog.py

```
@@ -500,118 +507,289 @@
+        if frontmatter.get("schema"):
+            frontmatter["schema"]["wordCount"] = word_count
+        
+        # 3. Ensure all images are absolute URLs
+        self._ensure_absolute_images(polished_data)
+        
+        # 4. Clean up any remaining AI placeholders
+        self._clean_ai_placeholders(polished_data, content)
+        
+        return polished_data
+
+    def _build_api_v3(self, data: Dict[str, Any], target_date: str, content: str, content_gen_frontmatter: Dict[str, Any] = None) -> Dict[str, Any]:
```

Ruff (0.12.2)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)